server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/SearchHighlightContext.java

```diff
@@ -20,8 +21,8 @@
 import java.util.Set;
 
 public class SearchHighlightContext {
-
     private final Map<String, Field> fields;
+    private QueryBuilder originalQuery;
```

Comment:
Can we make this final, and set in `HighlightBuilder.build()`?

Reply:
Done!